lib/internal/streams/pipeline.js

```diff
@@ -33,7 +36,18 @@ function destroyer(stream, reading, writing, callback) {
     closed = true;
   });
 
-  eos(stream, { readable: reading, writable: writing }, (err) => {
+  const opts = {
+    readable: reading,
+    writable: writing,
+  };
+  const cleanup = eos(stream, opts, (err) => {
+    // cleanup() must be called here to unregister the error handler
+    // that `eos` attaches to avoid swallowing the re-emitted error
+    //
+    // TODO: Should eos cleanup after itself?
+    // Ref https://github.com/mafintosh/end-of-stream/pull/12
+    cleanup();
```

Comment:
I think this should not be done here. `'error'` can still happen after `'end'`, making error handling more tricky.